src/routes/jira/sync/jira-sync-post.ts

```diff
@@ -59,3 +66,13 @@ const getStartTimeInDaysAgo = (commitsFromDate: Date | undefined) => {
 	if (commitsFromDate === undefined) return undefined;
 	return Math.floor((Date.now() -  commitsFromDate?.getTime()) / MILLISECONDS_IN_ONE_DAY);
 };
+
+const isIncrementalBackfilling = (subscription: Subscription, fromDate: Date | undefined): boolean => {
+	if (subscription.syncStatus === "FAILED") {
+		return false;
+	}
+	if (!fromDate) {
+		return false;
```

Comment:
/nit
not sure whether we should do it in this PR or another, probably should check the param and always make sure the `commitsFrom` is not empty.